pocs/utils/data.py

```diff
@@ -1,46 +1,145 @@
+        url = "http://data.astrometry.net/{}".format(fn)
+        try:
             df = data.download_file(url)
-            try:
-                shutil.move(df, dest)
-            except OSError as e:
-                print("Problem saving. (Maybe permissions?): {}".format(e))
+        except Exception as e:
+            if not self.keep_going:
+                raise e
+            print('Failed to download {}: {}'.format(url, e))
+            return False
+        try:
+            shutil.move(df, dest)
+            return True
+        except OSError as e:
+            if not self.keep_going:
+                raise e
+            print("Problem saving {}. Check permissions: {}".format(url, e))
+            return False
 
-    if wide_field:
-        for i in range(4110, 4119):
-            download_one_file('4100/index-{}.fits'.format(i))
 
-    if narrow_field:
-        for i in range(4210, 4219):
-            download_one_file('4200/index-{}.fits'.format(i))
+def download_all_files(data_folder=None, wide_field=True, narrow_field=False, keep_going=True):
+    dl = Downloader(
+        data_folder=data_folder,
+        keep_going=keep_going,
+        narrow_field=narrow_field,
+        wide_field=wide_field)
+    return dl.download_all_files()
 
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(
-        description=__doc__,
-        formatter_class=argparse.RawDescriptionHelpFormatter)
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument(
+        '--folder',
+        help=('Destination folder for astrometry indices. Default: {}'.format(
+            _default_data_folder())),
+        default=_default_data_folder())
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--keep_going',
+        action='store_true',
+        help='Ignore download failures and keep going to the other downloads (default)')
+    group.add_argument(
+        '--nokeep_going', action='store_true', help='Fail immediately if any download fails')
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--narrow_field', action='store_true', help='Do download narrow field indices')
+    group.add_argument(
+        '--nonarrow_field',
+        action='store_true',
+        help='Skip downloading narrow field indices (default)')
 
-    parser.add_argument('--folder', help='Folder to place astrometry data')
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        '--wide_field', action='store_true', help='Do download wide field indices (default)')
+    group.add_argument(
+        '--nowide_field', action='store_true', help='Skip downloading wide field indices')
 
     args = parser.parse_args()
 
     if args.folder and not os.path.exists(args.folder):
-        print("{} does not exist.".format(args.folder))
+        print("Warning, data folder {} does not exist.".format(args.folder))
```

Comment:
Warn or just auto-create? We create directories in a number of different places so might just make sense here.